test-app/tests/integration/components/headless-form-test.gts

```diff
@@ -0,0 +1,415 @@
+/* eslint-disable no-undef -- Until https://github.com/ember-cli/eslint-plugin-ember/issues/1747 is resolved... */
+/* eslint-disable simple-import-sort/imports,padding-line-between-statements,decorator-position/decorator-position -- Can't fix these manually, without --fix working in .gts */
+
+import { tracked } from '@glimmer/tracking';
+import { fn } from '@ember/helper';
+import { on } from '@ember/modifier';
+import {
+  click,
+  fillIn,
+  render,
+  rerender,
+  triggerEvent,
+} from '@ember/test-helpers';
+import { module, skip, test } from 'qunit';
+
+import HeadlessForm from 'ember-headless-form/components/headless-form';
+import sinon from 'sinon';
+import { setupRenderingTest } from 'test-app/tests/helpers';
+
+import type { RenderingTestContext } from '@ember/test-helpers';
+import type { InputType } from 'ember-headless-form';
+
+module('Integration Component headless-form', function (hooks) {
+  setupRenderingTest(hooks);
+
+  test('it renders form markup', async function (assert) {
+    await render(<template><HeadlessForm class="foo" novalidate /></template>);
+
+    assert.dom('form').exists('it renders as <form>');
+    assert
+      .dom('form')
+      .hasClass('foo', 'it accepts custom HTML classes')
+      .hasAttribute('novalidate', '', 'it accepts arbitrary HTML attributes');
+  });
+
+  module('form.field', function () {
+    test('form yields field component', async function (assert) {
+      const data = { firstName: 'Simon' };
+
+      await render(<template>
+        <HeadlessForm @data={{data}} as |form|>
+          <form.field @name="firstName">
+            <div data-test-user-content>foo</div>
+          </form.field>
+        </HeadlessForm>
+      </template>);
+
+      assert
+        .dom('[data-test-user-content]')
+        .exists('form field can render user content');
+
+      assert
+        .dom('form > [data-test-user-content]')
+        .exists('does not render anything on its own');
```

Comment:
Yeah, I see how this can be a bit confusing. The selector basically tests that there is no DOM node sitting between `<form>` and our `<div data-test-user-content>`, i.e. the `field` component does not add any markup on its own (which we want to do as little as possible, given were are "headless")

Reply:
Oh I see! Maybe it can be `field component contains no markup itself` or something then.

Reply:
Yeah, that's a _much_ better phrase, applied that. TY!